loader/windows/icd_windows.c

```diff
@@ -103,18 +156,57 @@ BOOL CALLBACK khrIcdOsVendorsEnumerate(PINIT_ONCE InitOnce, PVOID Parameter, PVO
                 KHR_ICD_TRACE("Value not zero, skipping\n");
                 continue;
             }
-
             // add the library
-            khrIcdVendorAdd(cszLibraryName);
+            AdapterAdd(cszLibraryName, ZeroLuid);
         }
+    }
 
-        result = RegCloseKey(platformsKey);
-        if (ERROR_SUCCESS != result)
+    // Add adapters according to DXGI's preference order
+    HMODULE hDXGI = LoadLibrary("dxgi.dll");
+    if (hDXGI)
+    {
+        IDXGIFactory* pFactory = NULL;
+        PFN_CREATE_DXGI_FACTORY pCreateDXGIFactory = (PFN_CREATE_DXGI_FACTORY)GetProcAddress(hDXGI, "CreateDXGIFactory");
+        HRESULT hr = pCreateDXGIFactory(&IID_IDXGIFactory, &pFactory);
```

Comment:
Is this guaranteed to be present on any version of Windows?
If so add a: KHR_ASSERT(pCreateDXGIFactory != NULL);